src/Rest/Routes/NewsRoute.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+/**
+ * The class register route for $className endpoint
+ *
+ * @package Unicorns\Rest\Routes
+ */
+
+declare(strict_types=1);
+
+namespace Unicorns\Rest\Routes;
+
+use Unicorns\Config\Config;
+use UnicornsVendor\EightshiftLibs\Helpers\Components;
+use UnicornsVendor\EightshiftLibs\Rest\Routes\AbstractRoute;
+use UnicornsVendor\EightshiftLibs\Rest\CallableRouteInterface;
+
+/**
+ * Class NewsRoute
+ */
+class NewsRoute extends AbstractRoute implements CallableRouteInterface
+{
+
+	/**
+	 * Method that returns project Route namespace.
+	 *
+	 * @return string Project namespace UnicornsVendor\for REST route.
+	 */
+	protected function getNamespace(): string
+	{
+		return Config::getProjectRoutesNamespace();
+	}
+
+	/**
+	 * Method that returns project route version.
+	 *
+	 * @return string Route version as a string.
+	 */
+	protected function getVersion(): string
+	{
+		return Config::getProjectRoutesVersion();
+	}
+
+	/**
+	 * Get the base url of the route
+	 *
+	 * @return string The base URL for route you are adding.
+	 */
+	protected function getRouteName(): string
+	{
+		return '/news';
```

Comment:
Why news ?
Can this route be used for more that just news ?
Why not something related to Load more so it can be used later for custom post type with additional parameters

Reply:
It's called news because it was meant to fetch news from that specific API.

I could (and will) make something more generic, but wouldn't it still have to be somewhat customized for each external API required?

Reply:
Yes it should and you should make it like that so it is customizable and extendable for multiple endpoints

Reply:
In this case, I have to disagree. If you were using graphql then you'd have one endpoint. REST loosely follows JSON:API standard, so endpoints should match the entity you are targeting. In this case, it's news, so it's ok to only target news. 

The load more functionality on the other hand can pick what endpoint to ping and handle the responses internally.

I would extract this string in a constant tho 😄  That way you can write integration tests where you can pick the route name from the constant 🙂

Reply:
I've reworked the load-more component to have choosable URL :)

Should I leave the endpoint as `/news` or make it more generic?

Reply:
I'm fine with leaving it